src/nnet3/nnet-simple-component.cc

```diff
@@ -2635,6 +2795,19 @@ int32 ConvolutionComponent::NumParameters() const {
   return (filter_params_.NumCols() + 1) * filter_params_.NumRows();
 }
 
+void ConvolutionComponent::Vectorize(VectorBase<BaseFloat> *params) const {
```

Comment:
I don't exactly know where Vectorize and Unvectorize are going to be used. But isn't it important to even add the hyper-parameters like filter and input dimensions to the params ?